### ISSUE TITLE
TypeScriptPropertyNameGenerator not respected when using the TypeScriptClientGenerator

### ISSUE THREAD

**Issue by thombrink** [RicoSuter/NSwag#3400]
When using a custom TypeScriptPropertyNameGenerator with the TypeScriptClientGenerator, it does execute, but does nothing with the result.

The TypeScriptGenerator on the other hand does respect the custom TypeScriptPropertyNameGenerator.

Is this expected behaviour?

**Comment by RicoSuter** [comment]
I think the TypeScriptPropertyNameGenerator is only used for DTO property names (TypeScriptGenerator) - i dont see why it would be used by the TypeScriptClientGenerator?

**Comment by thombrink** [comment]
Because I only want the DTO's of all api controllers. For this reason I set the GenerateDtoTypes property of the TypeScriptClientGeneratorSettings to true and call GenerateFile with the Contracts option.

![image](https://user-images.githubusercontent.com/5973204/113998041-f7fc8e80-9858-11eb-8f1a-3e3341e0c21e.png)

**Comment by thombrink** [comment]
I found the problem: if the TypeStyle enum is set to Interface, then the PropertyNameGenerator is not used to generate the property names.

So it has nothing to do with the diffrence between the TypeScriptClientGenerator and the TypeScriptGenerator!

**Comment by RicoSuter** [comment]
> Update: I saw that the Class.liquid uses PropertyName and Interface.liquid uses InterfaceName. I guess this is intended behaviour?

Yes this intended - in the case of interface we cannot rename properties as they have to match 1:1 with the server JSON - in the case of classes the server JSON can be mapped to new property names.
